packages/api-client/src/store/active-entities.ts

```diff
@@ -0,0 +1,189 @@
+    // TODO: Do we really need this? We have to handle undefined anyways
+    const collection =
+      collections[activeRouterParams.value.collection] ||
+      collections[activeWorkspace.value.collections[0]]
+
+    return requests[key] || requests[collection?.requests[0]]
+  })
+
+  /** Grabs the currently active example using the path param */
+  const activeExample = computed<RequestExample | undefined>(() => {
+    const key =
+      activeRouterParams.value[PathId.Examples] === 'default'
+        ? activeRequest.value?.examples[0] || ''
+        : activeRouterParams.value[PathId.Examples]
+
+    return requestExamples[key]
+  })
+
+  /**
+   * First collection that the active request is in
+   *
+   * TODO we should add collection to the route and grab this from the params
+   */
+  const activeCollection = computed(() => {
+    const requestUid = activeRequest.value?.uid
+    if (requestUid)
+      return Object.values(collections).find((c) =>
+        c.requests?.includes(requestUid),
+      )
+
+    const fallbackUid =
+      activeWorkspace.value.collections[0] ?? collections[0]?.uid
+
+    return collections[fallbackUid]
+  })
+
+  /** The currently selected server in the addressBar */
+  const activeServer = computed(
+    () =>
+      servers[
+        activeRequest.value?.selectedServerUid ||
+          activeCollection.value?.selectedServerUid ||
+          ''
+      ],
+  )
+
+  /** Cookie associated with the current route */
+  const activeCookieId = computed<string | undefined>(
+    () => activeRouterParams.value[PathId.Cookies],
+  )
+
+  /**
+   * Active list all available substitution variables. Server variables
+   * will be populated into the environment on spec loading
+   */
+  const activeEnvVariables = computed(() => {
+    if (!activeEnvironment.value) return []
+    // TODO: Must merge global variables and collection level variables here
+    // Return a list of key value pairs that includes dot nested paths
+    return flattenEnvVars(JSON.parse(activeEnvironment.value.value)).map(
+      ([key, value]) => ({
+        key,
+        value,
+      }),
+    )
+  })
+
+  return {
+    activeCollection,
+    activeCookieId,
+    activeExample,
+    activeRequest,
+    activeRouterParams,
+    activeEnvironment,
+    activeServer,
+    activeWorkspace,
+    activeWorkspaceCollections,
+    activeWorkspaceServers,
+    activeEnvVariables,
+    activeWorkspaceRequests,
+    router,
+  }
+}
+
+export type ActiveEntitiesStore = ReturnType<typeof createActiveEntitiesStore>
+export const ACTIVE_ENTITIES_SYMBOL =
+  Symbol() as InjectionKey<ActiveEntitiesStore>
+
+/**
+ * The active entities store
+ *
+ * This store returns anything related to the currently active entities
+ * The only reason this is a store and not a simple hook is due to storing the current router here
+ */
+export const useActiveEntities = (): ActiveEntitiesStore => {
+  const store = inject(ACTIVE_ENTITIES_SYMBOL)
```

Comment:
I actually did that at first but the issue was the router. I think since we have multiple routers going we ran into issues previously with the useRouter hook from vue/router. So then I just copied the store